src/pages/home/report/withReportAndReportActionOrNotFound.js

```diff
@@ -0,0 +1,153 @@
+
+export default function (WrappedComponent) {
+    const propTypes = {
+        /** The HOC takes an optional ref as a prop and passes it as a ref to the wrapped component.
+         * That way, if a ref is passed to a component wrapped in the HOC, the ref is a reference to the wrapped component, not the HOC. */
+        forwardedRef: PropTypes.func,
+
+        /** The report currently being looked at */
+        report: reportPropTypes,
+
+        /** Array of report actions for this report */
+        reportActions: PropTypes.shape(reportActionPropTypes),
+
+        /** The policies which the user has access to */
+        policies: PropTypes.objectOf(
+            PropTypes.shape({
+                /** The policy name */
+                name: PropTypes.string,
+
+                /** The type of the policy */
+                type: PropTypes.string,
+            }),
+        ),
+
+        /** Route params */
+        route: PropTypes.shape({
+            params: PropTypes.shape({
+                /** Report ID passed via route */
+                reportID: PropTypes.string,
+
+                /** ReportActionID passed via route */
+                reportActionID: PropTypes.string,
+            }),
+        }).isRequired,
+
+        /** Beta features list */
+        betas: PropTypes.arrayOf(PropTypes.string),
+
+        /** Indicated whether the report data is loading */
+        isLoadingReportData: PropTypes.bool,
+
+        /** Is the window width narrow, like on a mobile device? */
+        isSmallScreenWidth: PropTypes.bool.isRequired,
+    };
+
+    const defaultProps = {
+        forwardedRef: () => {},
+        reportActions: {},
+        report: {},
+        policies: {},
+        betas: [],
+        isLoadingReportData: true,
+    };
+
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportAndReportActionOrNotFound(props) {
+        const getReportAction = useCallback(() => {
+            let reportAction = props.reportActions[`${props.route.params.reportActionID}`];
+
+            // Handle threads if needed
+            if (reportAction === undefined || reportAction.reportActionID === undefined) {
+                reportAction = ReportActionsUtils.getParentReportAction(props.report);
+            }
+
+            return reportAction;
+        }, [props.report, props.reportActions, props.route.params.reportActionID]);
+
+        const reportAction = getReportAction();
+
+        // For small screen, we don't call openReport API when we go to a sub report page by deeplink
+        // So we need to call openReport here for small screen
+        useEffect(() => {
+            if (!props.isSmallScreenWidth || (!_.isEmpty(props.report) && !_.isEmpty(reportAction))) {
+                return;
+            }
+            Report.openReport(props.route.params.reportID);
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, []);
```

Comment:
We're only left with this now @dukenv0307? Do we need to disable the exhaustive-deps check? Why not add the `props.isSmallScreenWidth`, `props.report`, `props.route.params.reportID` to the dependencies?

Reply:
Are we sure? 

1. What happens when we're on Desktop/Web and resize the browser? and then perform the steps? 
2. Assume that we've got more than one Split bill requests URL shared on chat, and we click on both one after the other?

Reply:
Hey,
Sorry I should've been clear. I am not saying always need to add the dependencies. I wanted to know if it is required? If not, then we add a comment why certain properties are not required.

For example, 
- I feel `isSmallScreenWidth` is required
- I am not confident about `props.report`? 
- and I am pretty sure we don't need `route.params.reportID` as that would mean opening a new URL, which should probably reload the whole component, thereby calling `useEffect`

Please correct me if I my understanding is wrong? May be you could test at your end?

Reply:
thanks @dukenv0307. I'll test this in an hour.